glslc/src/file_compiler.cc

```diff
@@ -62,13 +62,46 @@ bool FileCompiler::CompileShaderFile(const std::string& input_file,
     source_string = string_piece(&(*input_data.begin()),
                                  &(*input_data.begin()) + input_data.size());
   }
-  StageDeducer deducer(input_file);
+
+  std::unique_ptr<FileIncluder> includer(new FileIncluder(&include_file_finder_));
+  options_.SetIncluder(std::move(includer));
+
+  shaderc::SpvModule result =
+      compiler_.CompileGlslToSpv(source_string.begin(), source_string.size(),
+                                 shader_stage, error_file_name.data(), options_);
+  total_errors_ += result.GetNumErrors();
+  total_warnings_ += result.GetNumWarnings();
+
   bool compilation_success =
-      Compile(source_string, shader_stage, error_file_name.str(), deducer,
-              glslc::FileIncluder(&include_file_finder_), output_stream,
-              &std::cerr, &total_warnings_, &total_errors_);
+      result.GetCompilationStatus() == shaderc_compilation_status_success;
+
+  // Handle the error message for failing to deduce the shader kind.
+  if (result.GetCompilationStatus() ==
+      shaderc_compilation_status_invalid_stage) {
+    if (IsGlslFile(error_file_name)) {
+      std::cerr << "glslc: error: "
+                << "'" << error_file_name << "': "
+                << ".glsl file encountered but no -fshader-stage "
+                   "specified ahead";
+    } else if (error_file_name == "<stdin>") {
+      std::cerr
+          << "glslc: error: '-': -fshader-stage required when input is from "
+             "standard "
+             "input \"-\"";
+    } else {
+      std::cerr << "glslc: error: "
+                << "'" << error_file_name << "': "
+                << "file not recognized: File format not recognized";
+    }
+    std::cerr << "\n";
+  }
```

Comment:
I guess the output stream is empty in that case, but you are right, it should return false before emitting output.